### PR TITLE
Implement clients page & components

### PR DESCRIPTION
**Pull request by RubenWerdmuller**

### CODE REVIEW

File: src/components/client-add/ClientAdd.jsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            />
+          </label>
+        </div>
+      </form>
+    </div>
+  );
+};
+
+ClientAdd.propTypes = {
+  addFormData: PropTypes.func.isRequired,
```
Comment (RobbinHabermehl) [inline src/components/client-add/ClientAdd.jsx]: `addFormData` sounds incorrect: you're not adding form data, but a client right?

File: src/components/client-add/ClientAdd.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  handleFormVisible: PropTypes.func.isRequired
```
Comment (RobbinHabermehl) [inline src/components/client-add/ClientAdd.jsx]: Consider names to be phrases: "handle form visible" would be incorrect then. Something like "handle form visibility" would be better. Also, `handle` suits events better. When toggling the visibility I'd name it `toggleFormVisibility`.

File: src/components/client/Client.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import IconArrowExpand from '../../assets/icons/icon-arrow-expand.svg';
+import styles from './Client.module.css';
+
+const Client = ({ client: { client } }) => {
```
Comment (RobbinHabermehl) [inline src/components/client/Client.jsx]: Having a `client` object in the `client` object seems to indicate the data structure isn't right.

File: src/components/client/Client.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const handleClick = () => setIsExpanded(!isExpanded);
+
+  const clientNumber = 'CLN_';
```
Comment (RobbinHabermehl) [inline src/components/client/Client.jsx]: Assuming this is mock data, add a comment before stating that.

File: src/components/clients/Clients.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          addFormData={addClient}
+          handleFormVisible={handleFormVisible}
+        />
+      )}
+      <div className={styles.header}>
+        <span className={styles.heading}>All Clients</span>
+        <button
+          className={`${styles.addMember}
+            ${isFormVisible && styles.addMemberVisible}`}
+          disabled={isFormVisible === true}
```
Comment (RobbinHabermehl) [inline src/components/clients/Clients.jsx]: You can use the shorthand notation here.

File: src/components/clients/Clients.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+          <IconPlus className={styles.addNewIcon} />
+          New Client
+        </button>
+        <button type="button" className={styles.sortMembers}>
+          Sort by:
+          <IconArrowDown />
+        </button>
+      </div>
+      {clients.map(client => {
+        return (
```
Comment (RobbinHabermehl) [inline src/components/clients/Clients.jsx]: Utilize the arrow function and lose the curly braces so you can omit the explicit `return`.

File: src/components/clients/Clients.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            <Client client={client} />
+          </React.Fragment>
+        );
+      })}
+    </React.Fragment>
+  );
+};
+
+Clients.propTypes = {
+  clients: PropTypes.arrayOf(
```
Comment (RobbinHabermehl) [inline src/components/clients/Clients.jsx]: Sort props alphabetically.

File: src/components/team-member-add/TeamMemberAdd.module.css [hunk 8]
```diff
@@ -1,9 +1,17 @@
...
 
+@media only screen and (min-width: 901px) {
+  .container {
+    display: block;
+  }
+}
+
 .header {
-  order: 99;
+  order: 9;
```
Comment (RobbinHabermehl) [inline src/components/team-member-add/TeamMemberAdd.module.css]: Do you really need the `order` property?

Comment (RubenWerdmuller) [reply src/components/team-member-add/TeamMemberAdd.module.css]: It's used to place the elements on top as header on desktop, but I did find a more elegant solution now!

File: src/components/client-add/ClientAdd.jsx [hunk 9]
```diff
@@ -30,18 +30,18 @@ const ClientAdd = ({ addFormData, handleFormVisible }) => {
...
-    setBranchAdress('');
     setPostalCode('');
+    toggleFormVisibility();
   };
 
   const handleSubmit = event => {
     event.preventDefault();
 
-    addFormData({
+    addClientData({
```
Comment (RobbinHabermehl) [inline src/components/client-add/ClientAdd.jsx]: Don't use abstract names like data, even not when preceded by a word like "client", it's simply "add client".

File: package.json [hunk 10]
```diff
@@ -8,9 +8,10 @@
     "build": "gatsby build",
     "develop": "gatsby develop",
     "format": "prettier --write \"**/*.{js,jsx,json,md}\"",
-    "start": "npm run develop",
+    "start": "npm run develop && npm:api",
```
Comment (RobbinHabermehl) [inline package.json]: Does this really work?

File: src/components/time-entry-add/TimeEntryAdd.jsx [hunk 11]
```diff
@@ -47,108 +47,111 @@ const TimeEntryAdd = ({ addFormData }) => {
...
-          <button
-            className={styles.addButton}
-            disabled={validity.form !== true}
-            type="submit"
-          >
-            Add
-          </button>
-        </form>
-      )}
+      <form
```
Comment (RobbinHabermehl) [inline src/components/time-entry-add/TimeEntryAdd.jsx]: Why did you remove the toggling behavior?

File: src/components/time-entry-add/TimeEntryAdd.jsx [hunk 12]
```diff
@@ -47,108 +47,111 @@ const TimeEntryAdd = ({ addFormData }) => {
...
+        className={` ${styles.form} ${isFormVisible && styles.formVisible}`}
```
Comment (RobbinHabermehl) [inline src/components/time-entry-add/TimeEntryAdd.jsx]: Remove the space.

File: src/redux/time-entries/selectors.js [hunk 13]
```diff
@@ -1,13 +1,27 @@
...
+  clientsItemsSelector,
+  ({ items }, clients) =>
+    items
+      .map(timeEntry => ({
+        ...timeEntry,
+        client: clients.find(client => client.id === timeEntry.client)
+          .companyName
+      }))
+      .sort(
+        (timeEntryA, timeEntryB) =>
```
Comment (RobbinHabermehl) [inline src/redux/time-entries/selectors.js]: For a `.sort()` the arguments `a` and `b` will suffice as that's very recognizable.

File: package.json (hunk 10 above)
Comment (RubenWerdmuller) [reply package.json]: It doesn't, yet!

File: src/components/time-entry-add/TimeEntryAdd.jsx (hunk 11 above)
Comment (RubenWerdmuller) [reply src/components/time-entry-add/TimeEntryAdd.jsx]: I removed this behavior only for desktop, since it should always be visible there.

File: package.json [hunk 14]
```diff
@@ -8,7 +8,7 @@
     "build": "gatsby build",
     "develop": "gatsby develop",
     "format": "prettier --write \"**/*.{js,jsx,json,md}\"",
-    "start": "npm run develop && npm:api",
+    "start": "npm run develop",
```
Comment (RobbinHabermehl) [inline package.json]: It would be nice though to start the API automatically:
`npm run develop & npm run api`

File: src/components/time-entries/TimeEntries.jsx [hunk 15]
```diff
@@ -52,21 +59,24 @@ const TimeEntries = ({
...
+  fetchTimeEntries: PropTypes.func.isRequired,
+  timeEntries: PropTypes.arrayOf(
+    PropTypes.shape({
+      startTimestamp: PropTypes.string
+    })
+  )
 };
 
 TimeEntries.defaultProps = {
+  clients: [],
```
Comment (RobbinHabermehl) [inline src/components/time-entries/TimeEntries.jsx]: Redux already provides initial values of, in this case, empty arrays.

File: src/components/time-entry-add/TimeEntryAdd.jsx (hunk 11 above)
Comment (RobbinHabermehl) [reply src/components/time-entry-add/TimeEntryAdd.jsx]: Like this it seems to be gone on mobile as well? You've removed the condition to show and hide the form.

File: src/components/time-entry-add/TimeEntryAdd.jsx [hunk 16]
```diff
@@ -65,19 +65,29 @@ const TimeEntryAdd = ({ addFormData }) => {
...
             onChange={({ target }) => setClient(target.value)}
             value={client}
             required
-          />
+          >
+            <option value="" disabled>
+              -- Select a client --
+            </option>
+            {clients.map(_client => {
+              return (
```
Comment (RobbinHabermehl) [inline src/components/time-entry-add/TimeEntryAdd.jsx]: Return the template immediately.

File: src/components/time-entry-add/TimeEntryAdd.jsx [hunk 17]
```diff
@@ -65,19 +65,29 @@ const TimeEntryAdd = ({ addFormData }) => {
...
+                <React.Fragment key={_client.id}>
```
Comment (RobbinHabermehl) [inline src/components/time-entry-add/TimeEntryAdd.jsx]: There's no need for a `Fragment` here.

File: src/components/time-entry-add/TimeEntryAdd.jsx [hunk 18]
```diff
@@ -167,7 +173,16 @@ const TimeEntryAdd = ({ addFormData }) => {
...
+  addFormData: PropTypes.func.isRequired,
+  clients: PropTypes.arrayOf(
+    PropTypes.shape({
+      companyName: PropTypes.string
+    })
+  )
+};
+
+TimeEntryAdd.defaultProps = {
+  clients: []
```
Comment (RobbinHabermehl) [inline src/components/time-entry-add/TimeEntryAdd.jsx]: There's no need for `defaultProps` like mentioned above.

File: package.json (hunk 14 above)
Comment (RubenWerdmuller) [reply package.json]: That is very nice.

File: src/components/time-entry-add/TimeEntryAdd.jsx (hunk 11 above)
Comment (RubenWerdmuller) [reply src/components/time-entry-add/TimeEntryAdd.jsx]: Two different buttons toggle the form visibility state. That state is used to trigger a CSS class in TimeEntryAdd called `formVisible`.

File: src/components/time-entries/TimeEntries.jsx [hunk 19]
```diff
@@ -60,6 +79,7 @@ const TimeEntries = ({
 
 TimeEntries.propTypes = {
   addTimeEntry: PropTypes.func.isRequired,
+  // eslint-disable-next-line react/require-default-props
```
Comment (RobbinHabermehl) [inline src/components/time-entries/TimeEntries.jsx]: Instead of disabling this ESLint rule I'd change the prop to `isRequired`.
